stamina-testkit/src/main/scala/stamina/testkit/StaminaTestKit.scala

```diff
@@ -0,0 +1,85 @@
+package stamina
+package testkit
+
+import scala.util._
+import scala.annotation.tailrec
+
+trait StaminaTestKit { self: org.scalatest.WordSpecLike ⇒
+
+  val defaultSampleId = "default"
+  case class PersistableSample(sampleId: String, persistable: AnyRef, description: Option[String]) {
+    override def toString = persistable.getClass.getSimpleName + description.map(" " + _).getOrElse("")
+  }
+
+  def sample(persistable: AnyRef) = new PersistableSample(defaultSampleId, persistable, None)
+  def sample(sampleId: String, persistable: AnyRef) = new PersistableSample(sampleId, persistable, Some(sampleId))
+  def sample(sampleId: String, persistable: AnyRef, description: String) = new PersistableSample(sampleId, persistable, Some(description))
+
+  implicit class TestablePersisters(persisters: Persisters) extends org.scalatest.Matchers {
+    def generateTestsFor(samples: PersistableSample*): Unit = {
+      samples.foreach { sample ⇒
+        performRoundtrip(sample)
+        deserializeStoredVersions(sample)
+      }
+      // Here we could verify test coverage
+    }
+
+    private def performRoundtrip(sample: PersistableSample) = {
+      s"persist and unpersist $sample" in {
+        persisters.canPersist(sample.persistable) should be(true)
+        persisters.unpersist(persisters.persist(sample.persistable)) should equal(sample.persistable)
+      }
+    }
+
+    @tailrec
+    private def deserializeStoredVersions(sample: PersistableSample, fromVersion: Int = 1): Unit = {
+
+      s"deserialize the stored serialized form of $sample version $fromVersion" in {
+        verifyByteStringDeserialization(sample, fromVersion)
+      }
+
+      if (fromVersion < latestVersion(sample.persistable))
+        deserializeStoredVersions(sample, fromVersion + 1)
+    }
+
+    def latestVersion(persistable: AnyRef) = persisters.persisters.find(_.canPersist(persistable)).map(_.currentVersion).max
```

Comment:
This is the first time I've seen anyone use the `max` function on an Option :smile:.

Unfortunately it throws `java.lang.UnsupportedOperationException: empty.max` if no persister is found so it might be good to perform the same `canPersist` assertion as in the roundtrip tests to prevent very confusing test failures and stacktraces; yes, the first test will also fail but it will fail with a nice neat error while this one will fail with the exception.

Alternatively, return the option and deal with it at the call site.